### DISCUSSION TITLE
Cancellation Token

### DISCUSSION THREAD

**Discussion by samael-sam** [eclipse-theia/theia#10621] [Q&A]
Hi,
I'm trying to understand Cancellation Token. Googling doesnt help. Any Documentation related to the same would be very much helpful. Thanks in advance.

**Comment by msujew** [comment]
Hi @samael-sam,

Our cancellation tokens are the same as used in vscode. See [here](https://code.visualstudio.com/api/references/vscode-api#cancellation-tokens) for some of their documentation on that.

There isn't a lot to cancellation tokens themselves, you can see the whole implementation of them [here](https://github.com/microsoft/vscode/blob/main/src/vs/base/common/cancellation.ts).
